app/models/building.rb

```diff
@@ -21,7 +22,19 @@ class Building < ApplicationRecord
   validates :bldrecnbr, uniqueness: true, presence: true
   validates :name, uniqueness: true
 
+  scope :active, -> { where(archived: false) }
+  scope :archived, -> { where(archived: true) }
+
   def full_address
     "#{address}, #{city}, #{state} #{zip}"
   end
+
+  def rooms
```

Comment:
why do we need custom rooms method?

Reply:
@srmoola  To get all rooms for a building. It's used in app/controllers/buildings_controller.rb:
```
def delete_building(building)
  Resource.where(room_id: building.rooms.ids).delete_all
  SpecificAttribute.where(room_id: building.rooms.ids).delete_all
  Room.where(floor_id: building.floors.ids).delete_all
  Floor.where(building_id: building).delete_all
  building.delete ? true : false
end
```